messaging/src/main/protos/Model.proto

```diff
@@ -0,0 +1,155 @@
+syntax = "proto3";
+
+package model;
+
+option java_package = "io.lantern.messaging";
+
+enum MessageDirection {
+  OUT = 0;
+  IN = 1;
+}
+
+enum ContactType {
+  DIRECT = 0; // e.g. a person
+  GROUP = 1;
+}
+
+// A globally unique identifier for a Contact
+message ContactId {
+  ContactType type = 1;
+  string id = 2; // the public IdentityKey for direct contacts, the globally unique group id (type 4 UUID) for groups, both base32 encoded
+}
+
+// A contact in the address book
+message Contact {
+  ContactId contactId = 1;
+  repeated string memberIds = 2; // the ids of group member Contacts, base32 encoded (only applies to Contacts of type GROUP)
+  string displayName = 3; // the display name shown for this Contact (specific to our current device, the contact may have different display names in other people's address books)
+  int64 createdTs = 4; // the millisecond unix timestamp when this contact was created
+  int64 mostRecentMessageTs = 5; // the millisecond unix timestamp of the most recent message with this contact
+  MessageDirection mostRecentMessageDirection = 6; // direction of the most recent message with this contact
+  string mostRecentMessageText = 7; // the text of the most recent message with this contact
+  string mostRecentAttachmentMimeType = 8; // the mime type of the most recent attachment with this contact
+  int32 messagesDisappearAfterSeconds = 9; // the number of seconds after which new messages will automatically disappear. <=0 means they won't disappear.
+}
+
+// An attachment to a Message
+message Attachment {
+  string mimeType = 1; // the MIME type for this attachment
+  bytes keyMaterial = 2; // 64 bytes of key material (32 bytes for AES256 encryption, 32 bytes for HmacSHA256 authentication)
+  bytes digest = 3; // secure digest of the attachment
+  int64 plaintextLength = 4; // the content length of the plain text
+  map<string, string> metadata = 5;
+  string downloadUrl = 6; // the URL from which the encrypted attachment can be downloaded
+}
+
+// A locally stored Attachment
+message StoredAttachment {
+  enum Status {
+    PENDING_ENCRYPTION = 0; // attachment has not yet been encrypted
+    PENDING_UPLOAD = 1; // attachment has not yet been uploaded/downloaded
```

Comment:
The comment is actually wrong, it only applies to uploads. However, PENDING_ENCRYPTION should really be `PENDING` since it does apply to downloads. I'll update.